src/tonic/api/collections_api.rs

```diff
@@ -39,16 +53,31 @@ impl Collections for CollectionsService {
     }
 }
 
+impl From<(Instant, CollectionsResponse)> for GetCollectionsResponse {
+    fn from(value: (Instant, CollectionsResponse)) -> Self {
+        let (timing, response) = value;
+        let collections = response
+            .collections
+            .into_iter()
+            .map(|desc| CollectionDescription { name: desc.name })
+            .collect::<Vec<_>>();
+        Self {
+            collections,
+            time: timing.elapsed().as_secs_f64(),
+        }
+    }
+}
+
 impl TryFrom<StorageOperations> for storage::content_manager::storage_ops::StorageOperations {
     type Error = Status;
 
     fn try_from(value: StorageOperations) -> Result<Self, Self::Error> {
         if let Some(create_collection) = value.create_collection {
             Ok(Self::try_from(create_collection)?)
-            // } else if let Some(update_collection) = self.update_collection {
-            //     Ok(update_colletion.into())
-            // } else if let Some(delete_collection) = self.delete_collection {
-            //     Ok(delete_colletion.into())
+        } else if let Some(update_collection) = value.update_collection {
+            Ok(update_collection.into())
+        } else if let Some(delete_collection) = value.delete_collection {
+            Ok(delete_collection.into())
         } else {
             Err(Status::failed_precondition(
```

Comment:
I was thinking that it might be the best approach to put all operations under the single data structure. In case of Rest API I was trying to make it simple to generate OpenAPI with only few endpoints, but turns out that clients generated in this way are over-complicated and do not support included structures well enough. So I would prefer to eventually make a separate endpoint for each type of queries. In case of gRPC it should be easier to start with separate structures, cause nobody depends on it yet.

Reply:
It would imply having a separate RPC for every collection operation, which makes sense. I found original design of the rest API a bit unusual as well. Will change

Reply:
Updated